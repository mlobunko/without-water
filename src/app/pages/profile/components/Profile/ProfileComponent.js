import React from 'react';
import { Alert } from 'reactstrap';
import Loader from 'shared/fetching/components/Loader';
import UserImage from '../UserImage';
import Social from '../Social';
import City from '../City';
import Languages from '../Languages';

export const ProfileComponent = ({
  fetching,
  isError,
  serverError,
  loginError
}) => (
  <React.Fragment>
    {fetching ? (
      <div className="container-vertical-60vh">
        <Loader />
      </div>
    ) : !isError ? (
      <div className="container-max-width profile">
        <UserImage />
        <City />
        <Languages />
        <Social />
      </div>
    ) : (
      <div className="container-vertical-60vh">
        <Alert color="danger">{serverError || loginError}</Alert>
      </div>
    )}
  </React.Fragment>
);
