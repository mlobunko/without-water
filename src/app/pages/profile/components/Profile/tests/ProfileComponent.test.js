import React from 'react';
import { shallow } from 'enzyme';

import { ProfileComponent } from '../ProfileComponent';

const initialProps = {
  city: 'London',
  fetching: true,
  isError: false,
  loginError: '',
  serverError: ''
};

describe('render', () => {
  it('render Loader', () => {
    const props = {
      ...initialProps,
      city: ''
    };
    const wrapper = shallow(<ProfileComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render StyledProfileComponent', () => {
    const props = {
      ...initialProps,
      fetching: false
    };
    const wrapper = shallow(<ProfileComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render Alert', () => {
    const props = {
      ...initialProps,
      fetching: false,
      city: '',
      isError: true,
      serverError: 'error'
    };
    const wrapper = shallow(<ProfileComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
