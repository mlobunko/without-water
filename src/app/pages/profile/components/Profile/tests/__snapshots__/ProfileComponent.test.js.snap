// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render render Alert 1`] = `
<Fragment>
  <Alert
    centerPage={true}
    color="danger"
    textError="error"
  />
</Fragment>
`;

exports[`render render Loader 1`] = `
<Fragment>
  <Component
    centerPage={true}
  />
</Fragment>
`;

exports[`render render StyledProfileComponent 1`] = `
<Fragment>
  <styled.div>
    <Component />
    <Connect(Component) />
    <Connect(Component) />
    <Connect(Component) />
  </styled.div>
</Fragment>
`;
